### ISSUE TITLE
Expires not respected on task retry

### ISSUE THREAD

**Issue by matt-snider** [celery/celery#3297]
Hi! Just as documented in #980 back in Sep 2012, it seems expires is not passed along when retrying. It appears this was fixed, but perhaps later re-implemented. Is this intentional?

As an example:

```python
@app.task(bind=True)
def add(self, x, y):
      # ...
      logger.info('Expires: {}'.format(self.request.expires))
      if some_condition:
           self.retry()
      return x + y

# somewhere else
add.apply_async((2, 2), expires=200)
```

In this case, the `self.request.expires` will only be set in the first attempt and not subsequent retries.

**Comment by matt-snider** [comment]
That was fast :open_mouth:... thanks!

**Comment by matt-snider** [comment]
Will this fix get backported into 3.1?
